flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -202,16 +204,27 @@ public YarnClusterDescriptor(
         this.nodeLabel = flinkConfiguration.getString(YarnConfigOptions.NODE_LABEL);
     }
 
-    private Optional<List<File>> decodeFilesToShipToCluster(
+    private Optional<List<Path>> decodeFilesToShipToCluster(
             final Configuration configuration, final ConfigOption<List<String>> configOption) {
         checkNotNull(configuration);
         checkNotNull(configOption);
 
-        final List<File> files =
-                ConfigUtils.decodeListFromConfig(configuration, configOption, File::new);
+        List<Path> files = ConfigUtils.decodeListFromConfig(configuration, configOption, Path::new);
+        files = files.stream().map(this::enrichPathSchemaIfNeeded).collect(Collectors.toList());
         return files.isEmpty() ? Optional.empty() : Optional.of(files);
     }
 
+    private Path enrichPathSchemaIfNeeded(Path path) {
+        if (isWithoutSchema(path)) {
+            return new Path(new File(path.toString()).toURI());
+        }
+        return path;
+    }
+
+    private boolean isWithoutSchema(Path path) {
+        return StringUtils.isNullOrWhitespaceOnly(path.toUri().getScheme());
+    }
```

Comment:
How about updating the `Path enrichPathSchemaIfNeeded(Path path)` to the `Path createPathWithSchema(String path)`?

If so, the `files = files.stream().map(this::enrichPathSchemaIfNeeded).collect(Collectors.toList());` can be removed. We just call `List<Path> files = ConfigUtils.decodeListFromConfig(configuration, configOption, this::createPathWithSchema);`

Reply:
@1996fanrui updated.